pkgs/development/ocaml-modules/sqlite3/default.nix

```diff
@@ -1,18 +1,27 @@
-{ stdenv, fetchurl, sqlite, ocaml, findlib, ocamlbuild, pkgconfig }:
+{ stdenv, fetchurl, sqlite, ocaml, findlib, ocamlbuild, pkgconfig, dune_2, dune-configurator, opaline }:
 
 stdenv.mkDerivation rec {
   pname = "ocaml-sqlite3";
-  version = "2.0.9";
+  version = "5.0.1";
 
   src = fetchurl {
-    url = "https://github.com/mmottl/sqlite3-ocaml/releases/download/v${version}/sqlite3-ocaml-${version}.tar.gz";
-    sha256 = "0rwsx1nfa3xqmbygim2qx45jqm1gwf08m70wmcwkx50f1qk3l551";
+    url = "https://github.com/mmottl/sqlite3-ocaml/releases/download/${version}/sqlite3-${version}.tbz";
+    sha256 = "0iymkszrs6qwak0vadfzc8yd8jfwn06zl08ggb4jr2mgk2c8mmsn";
   };
 
   nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ ocaml findlib ocamlbuild sqlite ];
-
-  createFindlibDestdir = true;
+  buildInputs = [ ocaml findlib ocamlbuild sqlite dune_2 ];
+  propagatedBuildInputs = [ dune-configurator ];
+  buildPhase = ''
+    runHook preBuild
+    dune build -p sqlite3
+    runHook postBuild
+  '';
+  installPhase = ''
+    runHook preInstall
+    ${opaline}/bin/opaline -prefix $out -libdir $OCAMLFIND_DESTDIR
+    runHook postInstall
+  '';
```

Comment:
What about using `buildDunePackage`?

Reply:
`buildDunePackage` passes the `pname` given on to dune, but the `pname` of this package (`ocaml-sqlite3`) doesn't match what dune needs to hear (`sqlite3`). I don't know nix that well so I just copied some bits from `buildDune2Package` and modified them to give dune a different value from the `pname`. There might be a better way to do it?

Reply:
This sounds like the `pname` is wrong. You may just fix it ;-)

Reply:
Ah, I was thinking the `pname` was affecting the name of the package in nix.